frontend/app/types.d.ts

```diff
@@ -63,23 +50,23 @@ declare module 'i18next' {
   interface CustomTypeOptions {
     defaultNS: false;
     resources: {
-      'address-validation': typeof addressValidation;
-      'apply-adult-child': typeof applyAdultChild;
-      'apply-adult': typeof applyAdult;
-      'apply-child': typeof applyChild;
-      apply: typeof apply;
-      'data-unavailable': typeof dataUnavailable;
-      gcweb: typeof gcweb;
-      index: typeof index;
-      letters: typeof letters;
-      status: typeof status;
-      renew: typeof renew;
-      'renew-ita': typeof renewIta;
-      'renew-child': typeof renewChild;
-      'renew-adult-child': typeof renewAdultChild;
-      'stub-login': typeof stubLogin;
-      'unable-to-process-request': typeof unableToProcessRequest;
-      'protected-renew': typeof protectedRenew;
+      'address-validation': enResources['addressValidation'] & frResources['addressValidation'];
+      'apply-adult-child': enResources['applyAdultChild'] & frResources['applyAdultChild'];
+      'apply-adult': enResources['applyAdult'] & frResources['applyAdult'];
+      'apply-child': enResources['applyChild'] & frResources['applyChild'];
+      apply: enResources['apply'] & frResources['apply'];
+      'data-unavailable': enResources['dataUnavailable'] & frResources['dataUnavailable'];
+      gcweb: enResources['gcweb'] & frResources['gcweb'];
+      index: enResources['index'] & frResources['index'];
+      letters: enResources['letters'] & frResources['letters'];
+      status: enResources['status'] & frResources['status'];
+      renew: enResources['renew'] & frResources['renew'];
+      'renew-ita': enResources['renewIta'] & frResources['renewIta'];
+      'renew-child': enResources['renewChild'] & frResources['renewChild'];
+      'renew-adult-child': enResources['renewAdultChild'] & frResources['renewAdultChild'];
+      'stub-login': enResources['stubLogin'] & frResources['stubLogin'];
+      'unable-to-process-request': enResources['unableToProcessRequest'] & frResources['unableToProcessRequest'];
+      'protected-renew': enResources['protectedRenew'] & frResources['protectedRenew'];
```

Comment:
This can be simplified and only uses `en` default locale.

Reply:
If one of the resource files needs an extra translation (looking at you, French **ಠ_ಠ**) then we need to merge both types.

It might be unlikely, but it has happened at least twice in the past in other applications I've worked on.

Reply:
Put emphasis on en. We can use tools like [i18next-locales-sync](https://github.com/felixmosh/i18next-locales-sync) to synchronize locales with en.

Reply:
Not sure you understand what I'm saying.
I've worked on projects in the past where a page might require two paragraphs of text for english, and 4 paragraphs of text for french.

In other words, the i18n files don't necessarily always sync.

Reply:
You return an array is this case.
https://www.i18next.com/translation-function/objects-and-arrays

Reply:
https://github.com/DTS-STN/canadian-dental-care-plan/pull/2747/commits/6a5c8fb12f3219aaa65f2ab19b00186e17134047

Reply:
Simplify it like this `resources: enResources & frResources;`. You won't need to add namespaces in `types.d.ts`.

Reply:
> Simplify it like this `resources: enResources & frResources;`. You won't need to add namespaces in `types.d.ts`.

Not sure what you mean here.